src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_SolarTower.java

```diff
@@ -551,45 +556,51 @@ private boolean addSolarHeater(IGregTechTileEntity aTileEntity, int a) {
 
 	@Override
 	public boolean checkRecipe(final ItemStack aStack) {
-		this.mEfficiencyIncrease = 10;
-		this.mMaxProgresstime = 100;
+		this.mEfficiencyIncrease = 100;
+		this.mMaxProgresstime = 200;
 
 		if (this.mSolarHeaters.isEmpty() || this.mSolarHeaters.size() < 340 || this.getTotalRuntimeInTicks() % 200 == 0) {
 			getConnectedSolarReflectors();			
 		}
 
 		int aTier = getHeaterTier();
 		int aHeaters = getHeaterCountForTier(aTier);
-		
-		//Manage Heat every 5s
-		//Add Heat First, if sources available
-		if (aHeaters > 0) {
-			for (int i = 0; i < aHeaters; i++) {
-				Math.min((this.mHeatLevel += aTier), 20000);
-			}
+
+		double aEfficiency;
+		// Original formula was (-Math.pow(this.mHeatLevel - 50000, 0.8) + 7000) / 7000
+		// However, negative numbers to the power of a non-integer result in NaN, by default
+		// Max efficiency is 1, at mHeatLevel = 50000, and it lowers at the same rate if going above or below this heat
+		// Min efficiency is 0.179, at mHeatLevel = 0 or 100000
+		if (this.mHeatLevel >= 50000) {
+			aEfficiency = (-Math.pow(this.mHeatLevel - 50000, 0.8) + 7000) / 7000;
 		}
-		
-		//Remove Heat, based on time of day
-		if (mHeatLevel > 0) {
-			if (mHeatLevel > 20000) {
-				this.mHeatLevel = 20000;
-			}
-		}		
+		else {
+			aEfficiency = (-Math.pow((100000 - this.mHeatLevel) - 50000, 0.8) + 7000) / 7000;
+		}
+
 		World w = this.getBaseMetaTileEntity().getWorld();
+
+		//Manage Heat every 5s
```

Comment:
5 seconds = 100 ticks. You increased to 200 ticks

Reply:
It was 5 seconds before, I forgot to change this comment that Alk had left before.